### PR TITLE
expression, executor, plan: rewrite builtin function unhex.

### PR DESCRIPTION
**Pull request by SteveZhangBit**

### CODE REVIEW

File: expression/builtin_string.go [hunk 1]
```diff
@@ -1061,49 +1061,54 @@ type unhexFunctionClass struct {
...
+	}
+
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf.tp.Flen = retFlen
+	types.SetBinChsClnFlag(bf.tp)
+	sig := &builtinUnHexSig{baseStringBuiltinFunc{bf}}
 	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
```
Comment (XuHuaiyu) [inline expression/builtin_string.go]: we may move errors.Trace(err) to line 1064 for checking args when we will use it in getFunction.

Comment (SteveZhangBit) [inline expression/builtin_string.go]: Sorry, I didn't get the idea. Did you mean that add 
`err := c.verifyArgs(args)`
`if err != nil { return nil, errors.Trace(err) }`
at line 1064

Comment (XuHuaiyu) [inline expression/builtin_string.go]: yes

Comment (SteveZhangBit) [inline expression/builtin_string.go]: fixed, and the same problem in hex function

**Comment by XuHuaiyu** [comment]
LGTM 
PTAL @zz-jason

**Pull request merged**